File: src/main/java/com/github/zametki/model/VerificationRecordType.java
package com.github.zametki.model;

import com.github.mjdbc.type.DbInt;
import org.jetbrains.annotations.NotNull;

public enum VerificationRecordType implements DbInt {
    EmailValidation(1),
    PasswordReset(2);

    private final int dbId;

    VerificationRecordType(int dbId) {
        this.dbId = dbId;
    }

    @NotNull
    public static VerificationRecordType fromId(int id) {
        if (id == EmailValidation.dbId) {
            return EmailValidation;
        } else if (id == PasswordReset.dbId) {
            return PasswordReset;
        }
        throw new IllegalArgumentException("Not valid id: " + id);
    }


    @Override
    public int getDbValue() {
        return dbId;
    }
}

File: src/main/java/com/github/zametki/model/VerificationRecordType.kt
package com.github.zametki.model

import com.github.mjdbc.type.DbInt

enum class VerificationRecordType(private val dbId: Int) : DbInt {
    EmailValidation(1),
    PasswordReset(2);


    override fun getDbValue() = dbId

    companion object {

        fun fromId(id: Int): VerificationRecordType {
            if (id == EmailValidation.dbId) {
                return EmailValidation
            } else if (id == PasswordReset.dbId) {
                return PasswordReset
            }
            throw IllegalArgumentException("Not valid id: " + id)
        }
    }
}
